axelor-project/src/main/java/com/axelor/apps/project/web/ProjectActivityDashboardController.java

```diff
@@ -29,18 +29,23 @@
 
   public void getData(ActionRequest request, ActionResponse response) {
     LocalDate todayDate = LocalDate.now();
+    Long projectId = (Long) request.getContext().get("id");
     response.setValues(
         Beans.get(ProjectActivityDashboardService.class)
-            .getData(todayDate.minusDays(29), todayDate));
+            .getData(todayDate.minusDays(29), todayDate, projectId));
   }
 
   public void getPreviousData(ActionRequest request, ActionResponse response) {
-    String fromDate = request.getContext().get("fromDate").toString();
-    response.setValues(Beans.get(ProjectActivityDashboardService.class).getPreviousData(fromDate));
+    String startDate = request.getContext().get("startDate").toString();
+    Long projectId = (Long) request.getContext().get("id");
+    response.setValues(
+        Beans.get(ProjectActivityDashboardService.class).getPreviousData(startDate, projectId));
   }
 
   public void getNextData(ActionRequest request, ActionResponse response) {
-    String toDate = request.getContext().get("toDate").toString();
-    response.setValues(Beans.get(ProjectActivityDashboardService.class).getNextData(toDate));
+    String endDate = request.getContext().get("endDate").toString();
+    Long projectId = (Long) request.getContext().get("id");
+    response.setValues(
+        Beans.get(ProjectActivityDashboardService.class).getNextData(endDate, projectId));
   }
```

Comment:
Please, like most controller, put everything in the method into a  try {...} catch (Exception e) { TracebackService.trace(response, e) } to trace any error